cli/azd/pkg/auth/manager.go

```diff
@@ -133,12 +138,7 @@ func EnsureLoggedInCredential(ctx context.Context, credential azcore.TokenCreden
 		Scopes: LoginScopes,
 	})
 	if err != nil {
-		// It is important that we dump the failure which contains error code, correlation IDs from AAD to log
-		// An improvement to make here is to classify 'unhandled' vs 'handled' errors
-		// where handled errors would be fixed with rerunning login (i.e. token expiry), vs.
-		// unhandled errors where it indicates a setup issue.
-		log.Printf("failed fetching access token: %s", err.Error())
-		return &azcore.AccessToken{}, ErrNoCurrentUser
+		return &azcore.AccessToken{}, err
```

Comment:
With this change, it's most likely that we'd catch token expiry type errors and return an error text: "re-authentication required, run `azd auth login` to login" based on the AD auth spec, but it's also possible that the spec of implementation is incorrect. In those cases, we'd end up showing the actual failure message from AD which may be helpful.

I think this is more technically correct, but we definitely could be showing unfriendly messages in edge cases where "not logged in, run `azd auth login` to login" would have been better.

Reply:
I tested it today, and it worked as expected. I tweaked the token expiry message slightly to read:

`login expired, run 'azd auth login' to log in`